### PR TITLE
update x-tyk-gw versions schema

### PR DESCRIPTION
**Pull request by jeffy-mathew**
[changelog]
internal: x-tyk-gw versions schema fix

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4078/merge
Commit:  
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2394991710)

**Review by furkansenharputlu — APPROVED** [review]

**Review by furkansenharputlu — APPROVED** [review]

**Review by urbanishimwe — APPROVED** [review]

File: apidef/oas/root.go [hunk 1]
```diff
@@ -123,7 +123,7 @@ type Versioning struct {
 	Default             string        `bson:"default" json:"default"`   // required
 	Location            string        `bson:"location" json:"location"` // required
 	Key                 string        `bson:"key" json:"key"`           // required
-	Versions            []VersionToID `bson:"versions" json:"versions"` // required
+	Versions            []VersionToID `bson:"versions" json:"versions"`
```
Comment (furkansenharputlu) [inline apidef/oas/root.go:126]: this should have `omitempty` if it is not required

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4078/merge
Commit: 489a074d4394f487491d9e2a06de14b900ff7581 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2395592460)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/4078/merge
Commit: 2e80b6aa6b20f66de598b1cc477f39a06830b1af 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2395599478)

**Pull request closed without merging**